Gordon360/Services/RecIM/MatchService.cs

```diff
@@ -168,16 +171,23 @@ public MatchExtendedViewModel GetMatchByID(int matchID)
                         {
                             ID = mt.TeamID,
                             Name = mt.Team.Name,
+                            Logo = mt.Team.Logo,
                             Status = mt.Status.Description,
                             Participant = mt.Team.ParticipantTeam
                                 .Where(pt => !new int[] {0,1,2}.Contains(pt.RoleTypeID)) //roletype is either deleted, invalid, invited to join
                                 .Select(pt => new ParticipantExtendedViewModel
                                 {
                                     Username = pt.ParticipantUsername,
-                                    Email = _accountService.GetAccountByUsername(pt.ParticipantUsername).Email,
+
+                                    FirstName = _participantService.GetParticipantFirstName(pt.ParticipantUsername),
+                                    LastName = _participantService.GetParticipantLastName(pt.ParticipantUsername),
+                                    IsCustom = _participantService.GetParticipantIsCustom(pt.ParticipantUsername),
+                                    Email = _context.Participant.First(p => p.Username == pt.ParticipantUsername).Email ?? _accountService.GetAccountByUsername(pt.ParticipantUsername).Email,
```

Comment:
There has to be a better way to do this, this is a horrible abuse of EF Core. You're calling four nested queries with their own nested queries for each participant. Perhaps we can make a `ParticipantService` method that takes a participant's username and returns a `ParticipantExtendedViewModel` instead of this mess?

Reply:
@Kevin-Chu58